fem/gridfunc.hpp

```diff
@@ -854,6 +855,60 @@ class ExtrudeCoefficient : public Coefficient
 GridFunction *Extrude1DGridFunction(Mesh *mesh, Mesh *mesh2d,
                                     GridFunction *sol, const int ny);
 
+/// The class provides methods for projecting function values evaluated on a
+/// set of points to a grid function. The values are directly copied to the
+/// nodal values of the target grid function if any of the points is matching
+/// a node of the grid function.
+class KDTreeNodalProjection
```

Comment:
In terms of organization, I think this class would better be defined in `kdtree.hpp`, in the similar spirit to the GSLIB interface residing almost entirely in `fem/gslib.hpp/.cpp`.

Reply:
It would also be good to explain how this class is used to deliver capabilities that the client might be looking for.  For example, this class uses k-d trees to move data defined on one MPI communicator to data structures on a different MPI communicator? IMHO, the Doxygen comments that are most helpful address what role this thing plays in the whole system? How can clients use it to achieve particular ends?

Reply:
gslib interpolates, on the other hand here we just map points. I will add interpolation extension later and I will use the gslib interface. 

I added additional comments.

Reply:
Can this class declaration be moved to `general/kdtree.hpp` with definitions in `general/kdtree.cpp`? I think it does not belong here alongside `GridFunction`.

Reply:
It cannot be in general as it acts on grid functions. After many discussions back in time - the conclusion was that the most appropriate place will be in the grid function file.